Tpetra:  link errors in Norms with deprecated code OFF

## Bug Report
@trilinos/tpetra 

### Description
Using the develop branch with Tpetra_ENABLE_DEPRECATED_CODE=OFF, I see the following link errors:
```
[ 50%] Linking CXX executable TpetraCore_BlockExpNamespace.exe
Undefined symbols for architecture x86_64:
  "void Tpetra::Details::normImpl<int, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace>, int>(int*, Kokkos::View<int const**, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::Serial, Kokkos::HostSpace> > const&, Tpetra::Details::EWhichNorm, Teuchos::ArrayView<unsigned long const> const&, bool, bool, Teuchos::Comm<int> const*)", referenced from:
      void (anonymous namespace)::multiVectorNormImpl<int, int, long long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace> >(Tpetra::MultiVector<int, int, long long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace> >::mag_type*, Tpetra::MultiVector<int, int, long long, Kokkos::Compat::KokkosDeviceWrapperNode<Kokkos::Serial, Kokkos::HostSpace> >&, Tpetra::Details::EWhichNorm) in libtpetra.a(Tpetra_MultiVector_INT_INT_LONG_LONG_SERIAL.cpp.o)
ld: symbol(s) not found for architecture x86_64

```

### Steps to Reproduce

```
cmake \
-D Trilinos_ENABLE_EXPLICIT_INSTANTIATION:BOOL=ON \
-D CMAKE_BUILD_TYPE:STRING="DEBUG" \
-D CMAKE_VERBOSE_MAKEFILE:BOOL=OFF \
\
-D MPI_BIN_DIR:PATH="/Users/kddevin/InstalledSoftware/openmpi-1.8.3_clang/bin" \
-D TPL_ENABLE_MPI:BOOL=ON \
-D MPI_EXEC_MAX_NUMPROCS:STRING=11 \
\
-D TPL_ENABLE_BinUtils:BOOL=OFF \
-D TPL_ENABLE_Pthread:BOOL=OFF \
\
-D CMAKE_C_FLAGS:STRING="-Wall  -pedantic -Wno-unknown-pragmas -Wno-narrowing -Wno-inline -Wshadow -Wdeprecated-declarations -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -fstrict-aliasing -Wno-long-long" \
-D CMAKE_CXX_FLAGS:STRING="-Wall -pedantic -Wno-unknown-pragmas -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-inline -Wshadow -Wdeprecated-declarations -Wempty-body  -Wignored-qualifiers -Wmissing-field-initializers  -Wsign-compare  -Wtype-limits   -Wuninitialized -Winit-self -fstrict-aliasing" \
\
-D Trilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=OFF \
-D Trilinos_ENABLE_TESTS:BOOL=OFF \
-D Trilinos_ENABLE_EXAMPLES:BOOL=OFF \
\
-D Trilinos_ENABLE_SHADOW_WARNINGS:BOOL=ON \
-D Trilinos_VERBOSE_CONFIGURE:BOOL=OFF \
-D Trilinos_ENABLE_Fortran:BOOL=OFF \
\
-D Trilinos_ENABLE_Tpetra:BOOL=ON \
-D Tpetra_ENABLE_TESTS:BOOL=ON \
-D Tpetra_ENABLE_EXAMPLES:BOOL=ON \
\
-D Tpetra_ENABLE_DEPRECATED_CODE:BOOL=OFF \
-D Tpetra_INST_INT_LONG_LONG:BOOL=ON \
-D Tpetra_INST_INT_INT:BOOL=OFF \
\
-D Teuchos_ENABLE_STACKTRACE:BOOL=OFF \
-D Teuchos_ENABLE_LONG_LONG_INT:BOOL=ON \
..
```

--- a/packages/tpetra/core/src/Tpetra_Details_normImpl.cpp
+++ b/packages/tpetra/core/src/Tpetra_Details_normImpl.cpp
@@ -50,11 +50,21 @@
 #include "Tpetra_Details_normImpl_def.hpp"
 #include "TpetraCore_ETIHelperMacros.h"
 
+
+#if defined(HAVE_TPETRA_INST_INT_INT)
+// don't need to do anything; Scalar=int is already added
+# define TPETRA_DETAILS_NORMIMPL_INSTANT_INT( NODE )
+#else 
+# define TPETRA_DETAILS_NORMIMPL_INSTANT_INT( NODE ) \
+  TPETRA_DETAILS_NORMIMPL_INSTANT( int, NODE )
+#endif 
+
 namespace Tpetra {
 
   TPETRA_ETI_MANGLING_TYPEDEFS()
 
   TPETRA_INSTANTIATE_SN( TPETRA_DETAILS_NORMIMPL_INSTANT )
+  TPETRA_INSTANTIATE_N( TPETRA_DETAILS_NORMIMPL_INSTANT_INT )
 
 #ifdef HAVE_TPETRA_INST_CUDA
 
@@ -66,6 +76,9 @@ namespace Tpetra {
   TPETRA_DETAILS_NORMIMPL_INSTANT( S, cuda_host_mirror_device_type )
 
   TPETRA_INSTANTIATE_S( TPETRA_DETAILS_NORMIMPL_INSTANT_CUDAHOSTMIRROR )
+#if ! defined(HAVE_TPETRA_INST_INT_INT)
+  TPETRA_DETAILS_NORMIMPL_INSTANT_CUDAHOSTMIRROR( int )
+#endif 
 
 #endif // HAVE_TPETRA_INST_CUDA
 

